### PR TITLE
TA-2914

### PR DESCRIPTION
**Pull request by vimalsuresh**
## What does this pull request do?

Add tests for disabling stage claims for following codes after 01/09/2018
IAAP => Asylum - Appeals (CLR)
IACA => Asylum - Stage 2a (CLR)
IACB => Asylum - Stage 2b (CLR)
IACC => Asylum - Stage 2c (CLR)
IACD => Asylum – Interim CLR rates (hourly rates with fixed fee for advocacy services)
IACE => Asylum – Stage 2d (CLR)
IACF => Asylum – Stage 2e (CLR)
IAXC => Asylum - CLR Work Not Subject to the GFS (CLR)
IMAP => Immigration - Appeals (CLR)
IMCA => Immigration - Stage 2a (CLR)
IMCB => Immigration - Stage 2b (CLR)
IMCC => Immigration - Stage 2c (CLR)
IMCD => Immigration – Interim CLR rates (hourly rates with fixed fee for advocacy services)
IMCE => Immigration – Stage 2d (CLR)
IMCF => Immigration – Stage 2e (CLR)
IMXC => Immigration - CLR Work Not Subject to the GFS (CLR)

Only Exception to above rule is IMXC:IBAI 

Disable stage disbursement claim for after 01/09/2018
IACA:IASY => Asylum - Stage 2a (CLR):First time asylum application
IACA:ILEA => Asylum - Stage 2a (CLR):Applications for leave to remain
IMCA:IDOM => Immigration - Stage 2a (CLR):Application for leave relating to Domestic Violence

## Why make these changes?

To test stage claims and stage disbursement claims are disabled for certain immigration and asylum codes

## Checklist

https://dsdmoj.atlassian.net/browse/TA-2914

### CODE REVIEW

**Review by solomonahmed — COMMENTED** [review]
Good work @vimalsuresh .  A few comments.  Can discuss when you're free if you prefer.

File: features/validations/legal_help/immigration_and_asylum/stage_disbursement_claims_disabled.feature [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+      | IACA:ILEA |
+      | IMCA:IDOM |
+    And the following outcomes are bulkloaded:
+      | # | CASE_ID | UFN        | CLAIM_TYPE | CASE_START_DATE | PROCUREMENT_AREA | ACCESS_POINT | EXEMPTION_CRITERIA_SATISFIED |
+      | 1 |     001 | 310818/001 | DC         |        31/08/18 | PA00142          | AP00187      | TR001                        |
+    Then user should see the outcome results page
+    And problem outcomes should equal 0
+    Then user confirms the submission
+
+  Scenario: Validate stage disbursement claims are enabled for certain Immigration and Asylum MT1 codes after 01/09/2018 , some codes error for other validation rules
```
[the remark this answers is not in the document]
Comment (vimalsuresh) [reply features/validations/legal_help/immigration_and_asylum/stage_disbursement_claims_disabled.feature:79]: split this test as the start date actives were different

File: features/validations/legal_help/immigration_and_asylum/stage_disbursement_claims_disabled.feature [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+      | IMCE:IOTH |
+      | IMCF:IOTH |
+    And the following outcomes are bulkloaded:
+      | # | CASE_ID | UFN        | CLAIM_TYPE | CASE_START_DATE | PROCUREMENT_AREA | ACCESS_POINT | EXEMPTION_CRITERIA_SATISFIED |
+      | 1 |     001 | 010423/001 | DC         |        01/04/23 | PA00142          | AP00187      | TR001                        |
+    Then user should see the outcome results page
+    And successful outcomes should equal 4
+    And problem outcomes should equal 0
+
+  Scenario: Validate stage disbursement claims are enabled for certain Immigration and Asylum MT1 codes irrespective of disabled date 01/09/2018 for certain codes, some do not have claim type DC
```
[the remark this answers is not in the document]
Comment (vimalsuresh) [reply features/validations/legal_help/immigration_and_asylum/stage_disbursement_claims_disabled.feature:18]: split the test into successful and not successful

File: features/validations/legal_help/immigration_and_asylum/stage_disbursement_claims_disabled.feature [hunk 3]
```diff
@@ -0,0 +1,105 @@
+Feature: Bulk load claims validations stage disbursement claims disabled for certain imiigration and Asylum codes after 01/09/2024
+
+  Scenario: Validate stage disbursement claims are enabled for certain Immigration and Asylum MT1 codes prior after 01/09/2018 using 01/04/23 as strat date because the start date active of these codes
```
[the remark this answers is not in the document]
Comment (vimalsuresh) [reply features/validations/legal_help/immigration_and_asylum/stage_disbursement_claims_disabled.feature:3]: amended

File: features/validations/legal_help/immigration_and_asylum/stage_disbursement_claims_disabled.feature [hunk 4]
```diff
@@ -0,0 +1,105 @@
+Feature: Bulk load claims validations stage disbursement claims disabled for certain imiigration and Asylum codes after 01/09/2024
```
[the remark this answers is not in the document]
Comment (vimalsuresh) [reply features/validations/legal_help/immigration_and_asylum/stage_disbursement_claims_disabled.feature:1]: amended

File: features/validations/legal_help/immigration_and_asylum/stage_claims_disabled.feature [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+      | IMCD:IOTH |
+      | IMCE:IOTH |
+      | IMCF:IOTH |
+      | IMXC:IOTH |
+    And the following outcomes are bulkloaded:
+      | # | CASE_ID | UFN        | CLAIM_TYPE | CASE_START_DATE | PROCUREMENT_AREA | ACCESS_POINT | EXEMPTION_CRITERIA_SATISFIED |
+      | 1 |     001 | 310818/001 | SC         |        31/08/18 | PA00142          | AP00187      | TR001                        |
+    Then user should see the outcome results page
+    And successful outcomes should equal 8
+    And problem outcomes should equal 8
```
[the remark this answers is not in the document]
Comment (vimalsuresh) [reply features/validations/legal_help/immigration_and_asylum/stage_claims_disabled.feature:80]: split the 4 more MT1 to other tests as start date is different. IACE, IMCE, IACF, IMCF is ignored for this test as the test is for prior to 01/09/18, but these codes have start date after 01/04/23.

**Review by solomonahmed — APPROVED** [review]

**Pull request merged**